js/apps/admin-ui/src/identity-providers/add/AddIdentityProvider.tsx

```diff
@@ -46,6 +46,17 @@ export default function AddIdentityProvider() {
     }
   }, [serverInfo, providerId]);
 
+  const showclientidsecret = () => {
+    const s = serverInfo.componentTypes?.["org.keycloak.broker.social.SocialIdentityProvider"]?.find(
+        ({ id }) => id === providerId,
+      );
+      if (s) {
+        return true;
+      } else {
+        return false;
+      }
+  };
```

Comment:
A couple of notes here:

- Rather than making this a function you can directly assign the result of your operation here to a variable.
- If you are looking to find if an item exists in an array you can use the [`some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method, which will return a boolean right away.
- We have a convention to use [camel case](https://en.wikipedia.org/wiki/Camel_case) for names of variables, so this would more appropriately be named `showClientIdSecret`.
- However, it is always better to name a variable after it's meaning rather than it's function. For example, this case we determine whether the IdP is a 'social provider', so it would more appropriately be named `isSocialIdentityProvider`.
- If you need to keep some variables in a function scope, or optimize performance you can do so in a [`useMemo()`](https://react.dev/reference/react/useMemo).

If you put all of that together you'll end up with something like:

```ts
const isSocialIdentityProvider = useMemo(() => {
  const namespace = "org.keycloak.broker.social.SocialIdentityProvider";
  const componentTypes = serverInfo.componentTypes?.[namespace] ?? [];

  return componentTypes.some(({ id }) => id === providerId);
}, [serverInfo, providerId]);
```